src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/DeallocationService.kt

```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.digital.hmpps.keyworker.services
+
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import uk.gov.justice.digital.hmpps.keyworker.domain.AllocationRepository
+import uk.gov.justice.digital.hmpps.keyworker.domain.ReferenceDataDomain.DEALLOCATION_REASON
+import uk.gov.justice.digital.hmpps.keyworker.domain.ReferenceDataRepository
+import uk.gov.justice.digital.hmpps.keyworker.domain.of
+import uk.gov.justice.digital.hmpps.keyworker.model.DeallocationReason
+
+@Transactional
+@Service
+class DeallocationService(
+  private val referenceDataRepository: ReferenceDataRepository,
+  private val allocationRepository: AllocationRepository,
+) {
+  fun deallocateExpiredAllocations(
+    prisonCode: String,
+    personIdentifier: String,
+    deallocationReason: DeallocationReason,
+  ) {
+    val reason = requireNotNull(referenceDataRepository.findByKey(DEALLOCATION_REASON of deallocationReason.reasonCode))
+    val allocations =
+      allocationRepository.findActiveForAllPolicies(personIdentifier).mapNotNull {
+        it.takeIf { it.prisonCode != prisonCode }?.apply { deallocate(reason) }
```

Comment:
There used to be a check that the destination prison code was a prison (`nomisService.isPrison(`) which means this would deallocate on transfers to hospitals, immigration centres etc. I think that's fine as a standard external movement to a TAP destination or court wouldn't be a transfer, release or arrival. Would you agree it's safe?